distributed/client.py

```diff
@@ -1053,6 +1057,26 @@ def _update_scheduler_info(self):
         except EnvironmentError:
             logger.debug("Not able to query scheduler for identity")
 
+    @gen.coroutine
+    def _wait_until_n(self, func, n):
+        while n and (yield func()) < n:
+            yield gen.sleep(0.2)
+
+    def wait_until_n(self, workers=0):
```

Comment:
I recommend the name `wait_for_workers` instead.  I'm not sure that a novice user will understand what `wait_until_n` means as immediately.